src/Common/Core/Impl/Extensions/IOExtensions.cs

```diff
@@ -100,8 +100,10 @@ public static class IOExtensions {
 
         public static bool HasReadPermissions(this string path) {
             try {
-                Directory.GetFiles(path);
-                return true;
+                if (Directory.Exists(path)) {
```

Comment:
`Directory.GetFiles` should've throw `IOException` and then you return false. Didn't it work?